addons/l10n_fr_pos_cert/static/src/js/pos.js

```diff
@@ -57,59 +57,83 @@ models.Order = models.Order.extend({
+        var selectedLine = order? order.selected_orderline: null;
+
+        return last_id === selectedLine.cid;
+    },
     set_quantity: function (quantity, keep_price) {
-        var current_quantity = this.get_quantity();
-        var new_quantity = parseFloat(quantity) || 0;
-        if (this.pos.is_french_country() && (new_quantity === 0 || new_quantity < current_quantity) && !this.reward_id) {
-            var quantity_to_decrease = current_quantity - new_quantity;
-            this.pos.gui.show_popup("number", {
-                'title': _t("Decrease the quantity by"),
-                'confirm': function (qty_decrease) {
-                    if (qty_decrease) {
-                        var order = this.pos.get_order();
-                        var selected_orderline = order.get_selected_orderline();
-                        qty_decrease = qty_decrease.replace(_t.database.parameters.decimal_point, '.');
-                        qty_decrease = parseFloat(qty_decrease);
-                        var decimals = this.pos.dp['Product Unit of Measure'];
-                        qty_decrease = round_di(qty_decrease, decimals);
-
-                        var current_total_quantity_remaining = selected_orderline.get_quantity();
-                        order.get_orderlines().forEach(function (orderline, index, array) {
-                            if (selected_orderline.id != orderline.id &&
-                                selected_orderline.get_product().id === orderline.get_product().id &&
-                                selected_orderline.get_discount() === orderline.get_discount()) {
-                                current_total_quantity_remaining += orderline.get_quantity();
-                            }
-                        });
+        var order = this.pos.get_order();
+        var selectedLine = order? order.selected_orderline: null;
 
-                        if (qty_decrease > current_total_quantity_remaining) {
-                          this.pos.gui.show_popup("error", {
-                              'title': _t("Order error"),
-                              'body':  _t("Not allowed to take back more than was ordered."),
-                          });
-                        } else {
-                            var decrease_line = selected_orderline.clone();
-                            decrease_line.order = order;
-                            orderline_super.set_quantity.apply(decrease_line, [-qty_decrease]);
-                            order.add_orderline(decrease_line);
-                        }
-                    }
-                }
-            });
+        if (this.pos.is_french_country() && selectedLine) {
+            var currentQuantity = selectedLine.get_quantity();
+            var newQuantity = parseFloat(quantity) || 0;
+
+            var last_id = Object.keys(order.orderlines._byId)[Object.keys(order.orderlines._byId).length-1];
+
+            if(newQuantity < currentQuantity) {
```

Comment:
en fait je viens de remarquer qu'on fait tout le temps appel à super sauf dans 1 seul cas, et je pense que ce qu t'as mis là est pas juste parce que t'as mis "newQuantity > 0" au lieu de "currentQuantity === 1", du coup on sait pas faire de négatif dirrectement.

Du coup on pourrait juste faire un if/else->
if (newQuantity < currentQuantity and not (this.isLastLine() and currentQuantity === 1)):
POPUP
else:
SUPER

Reply:
On a quand même un probleme. En gros on a 3 cas qui peuvent trigger le popup.
1) C'est pas la dernière ligne et la nouvelle quantité est plus petite.
2) C'est la denrière ligne, la currentQuantity = 1 et on entre une quantité < 1.
3) C'est la dernière ligne, currentQuantiy != 1 et newQuantiy < currentQuantity. 

Le problème dans le cas 3 c'est qu'à l'ajout d'un produit, newQuantity = 1 et current peut être égal à n'importe quoi, donc ça va toujours trigger. C'est pour ça que j'avais à la base mit le "new product" de mémoire et que finalement j'en avait plus besoin avec l'autre implémentation mais ça engendrait d'autres problemes.